### ISSUE TITLE
[P1] rem* ISA does not work correctly for negative numbers

### ISSUE THREAD

**Issue by marcinkoss** [tactcomplabs/rev#85]
Build and execute below code snippet
```
#define assert(x) if (!(x)) { asm(".dword 0x00000000"); }

int main() {
        int x = -20;
        int y = 6;
        int r = x % y;
        assert(r == -2);
        return 0;
}
```

Actual result in 'r' is '2' instead of '-2'.
That applies to all rem* ISA

**Comment by donofrio** [comment]
Confirmed we can reproduce on devel - fix in progress

**Comment by donofrio** [comment]
Fix and tests pushed to devel [849c15b](https://github.com/tactcomplabs/rev/commit/849c15b8849020b70c1a0cb3cab29d11b684dd00)

**Comment by marcinkoss** [comment]
Confirmed that it is working now.
